### PR TITLE
Remove hostnetworking dependency

### PR DESCRIPTION
**Pull request by Madhu-1**
Cephcsi RBD was using Hostnetworking due to the udev events required for the rbd map and unmap. with https://github.com/ceph/ceph-csi/pull/1450 we don't need hostNetworking for RBD anymore. 
However, CephFS works with pod networking.

### CODE REVIEW

**Review by nixpanic — COMMENTED** [review]
the 1st commit has an incorrect format, the Signed-off-by line needs to be split by an empty line...

**Review by nixpanic — APPROVED** [review]

**Comment by Madhu-1** [comment]
restarted CI due to E2E timeout issue

**Review by humblec — APPROVED** [review]

**Comment by nixpanic** [comment]
ci/centos/mini-e2e/k8s-1.19 failed with #1795 :

```
Feb 12 07:33:14.210: FAIL: failed to validate encrypted pvc with error timed out waiting for the condition
```

#1863 has just been merged, so this PR can be rebased and the CI issue hopefully prevented.

**Comment by nixpanic** [comment]
@mergifyio rebase

**Comment by nixpanic** [comment]
Both jobs seem to have gotten stuck at `STEP: ensuring all operations will work within a rados namespace`:

```
Feb 12 13:03:55.782: INFO: Running '/usr/bin/kubectl --server=https://192.168.39.11:8443 --kubeconfig=/root/.kube/config --namespace=cephcsi-e2e-84cac80d delete po -l app in (ceph-csi-rbd, csi-rbdplugin, csi-rbdplugin-provisioner) --ignore-not-found=false --namespace=cephcsi-e2e-84cac80d'
```

```
Feb 12 13:11:57.310: INFO: Running '/usr/bin/kubectl --server=https://192.168.39.204:8443 --kubeconfig=/root/.kube/config --namespace=cephcsi-e2e-d79bc147 delete po -l app in (ceph-csi-rbd, csi-rbdplugin, csi-rbdplugin-provisioner) --ignore-not-found=false --namespace=cephcsi-e2e-d79bc147'
```

**Comment by nixpanic** [comment]
Both **ci/centos/mini-e2e** tests hit a timeout after 30-50 minutes. The timeout is set to 90 minutes, so the last part of the waiting did not log any messages. This suggests that a `kubectl` command got stuck:
```
goroutine 15 [select, 43 minutes]:
k8s.io/kubernetes/test/e2e/framework.KubectlBuilder.ExecWithFullOutput(0xc0003ac000, 0x0, 0xc000446848, 0x6, 0x6, 0xc000b060a0, 0xc001704080, 0xc0007aa000)
	/go/src/github.com/ceph/ceph-csi/vendor/k8s.io/kubernetes/test/e2e/framework/util.go:636 +0x445
k8s.io/kubernetes/test/e2e/framework.KubectlBuilder.Exec(...)
	/go/src/github.com/ceph/ceph-csi/vendor/k8s.io/kubernetes/test/e2e/framework/util.go:618
k8s.io/kubernetes/test/e2e/framework.RunKubectl(0x7ffc358f1da0, 0x14, 0xc000446848, 0x6, 0x6, 0xc001704080, 0x20, 0x0, 0x0)
	/go/src/github.com/ceph/ceph-csi/vendor/k8s.io/kubernetes/test/e2e/framework/util.go:665 +0x71
github.com/ceph/ceph-csi/e2e.deletePodWithLabel(0x1c72652, 0x3f, 0x7ffc358f1da0, 0x14, 0xc0004c7300, 0x0, 0x0)
	/go/src/github.com/ceph/ceph-csi/e2e/pod.go:215 +0x229
```
and
```
goroutine 11 [select, 51 minutes]:
k8s.io/kubernetes/test/e2e/framework.KubectlBuilder.ExecWithFullOutput(0xc00003eb00, 0x0, 0xc000c64848, 0x6, 0x6, 0xc0000bd3f0, 0xc000a02a40, 0xc00045e000)
	/go/src/github.com/ceph/ceph-csi/vendor/k8s.io/kubernetes/test/e2e/framework/util.go:636 +0x445
k8s.io/kubernetes/test/e2e/framework.KubectlBuilder.Exec(...)
	/go/src/github.com/ceph/ceph-csi/vendor/k8s.io/kubernetes/test/e2e/framework/util.go:618
k8s.io/kubernetes/test/e2e/framework.RunKubectl(0x7ffe449cada0, 0x14, 0xc000c64848, 0x6, 0x6, 0xc000a02a40, 0x20, 0x0, 0x0)
	/go/src/github.com/ceph/ceph-csi/vendor/k8s.io/kubernetes/test/e2e/framework/util.go:665 +0x71
github.com/ceph/ceph-csi/e2e.deletePodWithLabel(0x1c72652, 0x3f, 0x7ffe449cada0, 0x14, 0xc000533600, 0x0, 0x0)
	/go/src/github.com/ceph/ceph-csi/e2e/pod.go:215 +0x229
```

Maybe we need to increase the resources for the minikube VM?

**Comment by nixpanic** [comment]
> /retest ci/centos/mini-e2e

#1867 has been merged, and the jobs in Jenkins updated. Maybe with more RAM for the minikube VM things pass again?

**Comment by Madhu-1** [comment]
@mergifyio rebase

**Comment by Madhu-1** [comment]
>panic: test timed out after 1h30m0s

CI failed with timeout error

**Comment by nixpanic** [comment]
This PR seems to be the only one that hits the timeouts, and only on the non-helm deployment. Maybe there are some issues with host-networking?

**Comment by Madhu-1** [comment]
@nixpanic but there is no difference between the helm and kubectl deployment. yes we need to verify where exectly most of the time is spent

**Review by idryomov — APPROVED** [review]
Glad to see this!

**Comment by Madhu-1** [comment]
From [ci/centos/mini-e2e/k8s-1.19](https://jenkins-ceph-csi.apps.ocp.ci.centos.org/blue/rest/organizations/jenkins/pipelines/mini-e2e_k8s-1.19/runs/732/nodes/85/steps/88/log/?start=0)  the test started around 10:03 and panicked at 10:28:13 this is hardly 26 minutes and the test timeout is around 90 minutes.

From: [ci/centos/mini-e2e/k8s-1.20](https://jenkins-ceph-csi.apps.ocp.ci.centos.org/blue/rest/organizations/jenkins/pipelines/mini-e2e_k8s-1.20/runs/286/nodes/85/steps/88/log/?start=0)  the test started around 10:02 and panicked at 10:42 this is hardly 42 minutes and the test timeout is around 90 minutes.

Not sure what is exactly happening in E2E @nixpanic any idea is this timeout due to the Jenkins?

**Comment by humblec** [comment]
Lgtm. @Madhu-1 I feel, its good to mention it in the release too! . Thanks!

**Comment by Madhu-1** [comment]
> Lgtm. @Madhu-1 I feel, its good to mention it in the release too! . Thanks!

sure will take care of it during release time.

**Review by humblec — APPROVED** [review]

**Comment by nixpanic** [comment]
> From [ci/centos/mini-e2e/k8s-1.19](https://jenkins-ceph-csi.apps.ocp.ci.centos.org/blue/rest/organizations/jenkins/pipelines/mini-e2e_k8s-1.19/runs/732/nodes/85/steps/88/log/?start=0) the test started around 10:03 and panicked at 10:28:13 this is hardly 26 minutes and the test timeout is around 90 minutes.
> 
> From: [ci/centos/mini-e2e/k8s-1.20](https://jenkins-ceph-csi.apps.ocp.ci.centos.org/blue/rest/organizations/jenkins/pipelines/mini-e2e_k8s-1.20/runs/286/nodes/85/steps/88/log/?start=0) the test started around 10:02 and panicked at 10:42 this is hardly 42 minutes and the test timeout is around 90 minutes.
> 
> Not sure what is exactly happening in E2E @nixpanic any idea is this timeout due to the Jenkins?

No, this isnt due to Jenkins. If there is a Jenkins issue, there will not be a golang panic.

My guess is that the last command got stuck, and did not log any output afterwards:

```
Feb 19 10:28:12.225: INFO: ExecWithOptions {Command:[/bin/sh -c rbd namespace create --pool=newrbdpool --namespace e2e-ns] Namespace:rook-ceph PodName:rook-ceph-tools-5949d6759-p9589 ContainerName:rook-ceph-tools Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:true Quiet:false}
Feb 19 10:28:12.225: INFO: >>> kubeConfig: /root/.kube/config
Feb 19 10:28:13.117: INFO: Running '/usr/bin/kubectl --server=https://192.168.39.76:8443 --kubeconfig=/root/.kube/config --namespace=cephcsi-e2e-499dce43 delete po -l app in (ceph-csi-rbd, csi-rbdplugin, csi-rbdplugin-provisioner) --ignore-not-found=false --namespace=cephcsi-e2e-499dce43'
```

You can check the stages of the Jenkins job, and see that the step running e2e took 1h 30m 14s at https://jenkins-ceph-csi.apps.ocp.ci.centos.org/blue/organizations/jenkins/mini-e2e_k8s-1.19/detail/mini-e2e_k8s-1.19/732/pipeline

**Review by nixpanic — DISMISSED** [review]

**Comment by Madhu-1** [comment]
> > From [ci/centos/mini-e2e/k8s-1.19](https://jenkins-ceph-csi.apps.ocp.ci.centos.org/blue/rest/organizations/jenkins/pipelines/mini-e2e_k8s-1.19/runs/732/nodes/85/steps/88/log/?start=0) the test started around 10:03 and panicked at 10:28:13 this is hardly 26 minutes and the test timeout is around 90 minutes.
> > From: [ci/centos/mini-e2e/k8s-1.20](https://jenkins-ceph-csi.apps.ocp.ci.centos.org/blue/rest/organizations/jenkins/pipelines/mini-e2e_k8s-1.20/runs/286/nodes/85/steps/88/log/?start=0) the test started around 10:02 and panicked at 10:42 this is hardly 42 minutes and the test timeout is around 90 minutes.
> > Not sure what is exactly happening in E2E @nixpanic any idea is this timeout due to the Jenkins?
> 
> No, this isnt due to Jenkins. If there is a Jenkins issue, there will not be a golang panic.
> 
> My guess is that the last command got stuck, and did not log any output afterwards:
> 
> ```
> Feb 19 10:28:12.225: INFO: ExecWithOptions {Command:[/bin/sh -c rbd namespace create --pool=newrbdpool --namespace e2e-ns] Namespace:rook-ceph PodName:rook-ceph-tools-5949d6759-p9589 ContainerName:rook-ceph-tools Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:true Quiet:false}
> Feb 19 10:28:12.225: INFO: >>> kubeConfig: /root/.kube/config
> Feb 19 10:28:13.117: INFO: Running '/usr/bin/kubectl --server=https://192.168.39.76:8443 --kubeconfig=/root/.kube/config --namespace=cephcsi-e2e-499dce43 delete po -l app in (ceph-csi-rbd, csi-rbdplugin, csi-rbdplugin-provisioner) --ignore-not-found=false --namespace=cephcsi-e2e-499dce43'
> ```
> 
> You can check the stages of the Jenkins job, and see that the step running e2e took 1h 30m 14s at https://jenkins-ceph-csi.apps.ocp.ci.centos.org/blue/organizations/jenkins/mini-e2e_k8s-1.19/detail/mini-e2e_k8s-1.19/732/pipeline


Normal   Killing        18m                 kubelet            Stopping container csi-rbdplugin
  Warning  FailedKillPod  16m                 kubelet            error killing pod: failed to "KillContainer" for "csi-rbdplugin" with KillContainerError: "rpc error: code = Unknown desc = operation timeout: context deadline exceeded"
  Warning  FailedSync     34s (x51 over 14m)  kubelet            error determining status: rpc error: code = DeadlineExceeded desc = context deadline exceeded

the pods are not getting deleted and it's getting timeout.

File: build.env [hunk 1]
```diff
@@ -36,7 +36,7 @@ SNAPSHOT_VERSION=v3.0.1
 HELM_VERSION=v3.1.2
 
 # minikube settings
-MINIKUBE_VERSION=v1.17.1
+MINIKUBE_VERSION=v1.18.0
```
Comment (humblec) [inline build.env]: @Madhu-1 do we need to update minikube with this change?

Comment (Madhu-1) [reply build.env]: @humblec am updating minikube to check is there any issue in the kubernetes network plugins  as the tests are timing out

**Comment by Madhu-1** [comment]
>STEP 3: ENV  GOPATH=${GOPATH}  GO111MODULE=on  PATH="${GOPATH}/bin:/opt/commitlint/node_modules/.bin:${PATH}"
Error: error checking if cached image exists from a previous build: error getting history of "da29eb791639286e59fda12e156b0da42ba1b617aff42f8ba4d81b24b39fd565": error creating new image from reference to image "da29eb791639286e59fda12e156b0da42ba1b617aff42f8ba4d81b24b39fd565": error locating item named "manifest" for image with ID "da29eb791639286e59fda12e156b0da42ba1b617aff42f8ba4d81b24b39fd565": file does not exist
make: *** [Makefile:207: .test-container-id] Error 125

CI failing with some error restarting the test.

**Comment by nixpanic** [comment]
The **ci/centos/mini-e2e** tests that timeout always hang at

    Running '/usr/bin/kubectl --server=https://192.168.39.26:8443 --kubeconfig=/root/.kube/config --namespace=cephcsi-e2e-4ea5ed5f delete po -l app in (ceph-csi-rbd, csi-rbdplugin, csi-rbdplugin-provisioner) --ignore-not-found=false --namespace=cephcsi-e2e-4ea5ed5f'

This command looks very weird, but it is what is specified in `e2e/rbd.go` around line 1157:

```go
                                        // delete csi pods
                                        err = deletePodWithLabel("app in (ceph-csi-rbd, csi-rbdplugin, csi-rbdplugin-provisioner)",
                                                cephCSINamespace, false)
```

Unfortunately I do not see why this command would depend on the network changes...

**Comment by Madhu-1** [comment]
> The **ci/centos/mini-e2e** tests that timeout always hang at
> 
> ```
> Running '/usr/bin/kubectl --server=https://192.168.39.26:8443 --kubeconfig=/root/.kube/config --namespace=cephcsi-e2e-4ea5ed5f delete po -l app in (ceph-csi-rbd, csi-rbdplugin, csi-rbdplugin-provisioner) --ignore-not-found=false --namespace=cephcsi-e2e-4ea5ed5f'
> ```
> 
> This command looks very weird, but it is what is specified in `e2e/rbd.go` around line 1157:
> 
> ```go
>                                         // delete csi pods
>                                         err = deletePodWithLabel("app in (ceph-csi-rbd, csi-rbdplugin, csi-rbdplugin-provisioner)",
>                                                 cephCSINamespace, false)
> ```
> 
> Unfortunately I do not see why this command would depend on the network changes...

something is blocking the pods to get deleted when we issue the command

```Warning FailedKillPod 16m kubelet error killing pod: failed to "KillContainer" for "csi-rbdplugin" with KillContainerError: "rpc error: code = Unknown desc = operation timeout: context deadline exceeded"
Warning FailedSync 34s (x51 over 14m) kubelet error determining status: rpc error: code = DeadlineExceeded desc = context deadline exceeded
```
this is something happening at the kubelet level. could be an issue in CRI or CNI or something else. I was just trying to isolate the issue.

**Comment by Madhu-1** [comment]
it happens with standalone template deployment, not with the helm charts

**Comment by nixpanic** [comment]
> > STEP 3: ENV  GOPATH=${GOPATH}  GO111MODULE=on  PATH="${GOPATH}/bin:/opt/commitlint/node_modules/.bin:${PATH}"
> > Error: error checking if cached image exists from a previous build: error getting history of "da29eb791639286e59fda12e156b0da42ba1b617aff42f8ba4d81b24b39fd565": error creating new image from reference to image "da29eb791639286e59fda12e156b0da42ba1b617aff42f8ba4d81b24b39fd565": error locating item named "manifest" for image with ID "da29eb791639286e59fda12e156b0da42ba1b617aff42f8ba4d81b24b39fd565": file does not exist
> > make: *** [Makefile:207: .test-container-id] Error 125
> 
> CI failing with some error restarting the test.

This is something recent, and should get addressed in the CI with #1900

File: build.env (hunk 1 above)
Comment (humblec) [reply build.env]: Oh. .Ok.. thanks for the confirmation

**Review by humblec — DISMISSED** [review]

**Comment by nixpanic** [comment]
> it happens with standalone template deployment, not with the helm charts

Any idea why this is still the case?

**Comment by Madhu-1** [comment]
> > it happens with standalone template deployment, not with the helm charts
> 
> Any idea why this is still the case?

Checking on this one.

**Comment by Madhu-1** [comment]
Added DNM doing some experimentation

**Comment by Madhu-1** [comment]
We have seen issues related to plugin pod restart when using pod networking. Not needed now.

**Pull request closed without merging**